File: mvn-advent-of-code/src/test/java/advent/twenty_seventeen/Day24ElectromagneticMoatTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day24ElectromagneticMoatTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day24ElectromagneticMoat("/twenty_seventeen/Day24-ElectromagneticMoat-input.txt");

        assertEquals(1511, dp.getPart1Answer().intValue());
        assertEquals(1471, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day24ElectromagneticMoat("/twenty_seventeen/Day24-ElectromagneticMoat-example#1.txt");

        assertEquals(31, dp.getPart1Answer().intValue());
        assertEquals(19, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day24ElectromagneticMoatTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day24ElectromagneticMoatTest  extends FlatSpec with Matchers {
  "2017 Day 24 - Input" should "calculate answers" in {
    val dp = new Day24ElectromagneticMoat("/twenty_seventeen/Day24-ElectromagneticMoat-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day24ElectromagneticMoat("/twenty_seventeen/Day24-ElectromagneticMoat-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
